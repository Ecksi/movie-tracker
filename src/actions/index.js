export const addMovies = movies => ({
  type: 'ADD_MOVIES',
  movies
});

export const toggleLogin = () => ({
  type: 'TOGGLE_LOGIN'
});
