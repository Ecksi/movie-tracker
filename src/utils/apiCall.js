import apiKey from '../private/apiKey';
import { doFetch}  from './doFetch';

const getMovieData = async () => {
  const url = 'https://api.themoviedb.org/3/movie/now_playing?api_key=';

  return await doFetch(url+apiKey);
};

export default getMovieData;