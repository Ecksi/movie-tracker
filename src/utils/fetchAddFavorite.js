export const fetchAddFavorite = async (movie, user_id) => {
  const url = 'http://localhost:3000/api/users/favorites/new';
  const options = {
    method: 'POST',
    body: JSON.stringify({
      ...movie,
      user_id
    }),
    headers: { 'Content-Type': 'application/json' }
  };
  return await fetch(url, options);
};
