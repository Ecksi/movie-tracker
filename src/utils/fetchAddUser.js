export const fetchAddUser = async user => {
  const url = 'http://localhost:3000/api/users/new';
  const options = {
    method: 'POST',
    body: JSON.stringify(user),
    headers: { 'Content-Type': 'application/json' }
  };
  await fetch(url, options);
};
