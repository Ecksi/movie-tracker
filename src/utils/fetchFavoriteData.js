import { doFetch } from './doFetch';

export const fetchFavorites = async user_id => {
  const url = `http://localhost:3000/api/users/${user_id}/favorites`;
  const favoritesList = await doFetch(url);
  return favoritesList.data;
};

