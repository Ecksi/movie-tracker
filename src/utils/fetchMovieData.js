import apiKey from '../private/apiKey';
// import { doFetch } from './doFetch';

// export const fetchMovieData = async () => {
//   const url = 'https://api.themoviedb.org/3/movie/now_playing?api_key=';
//   const dirtyMovieData = await doFetch(url + apiKey);
//   return cleanMovieData(dirtyMovieData.results);
// };

// const cleanMovieData = dirtyMovieData => {
//   const cleanMovieData = dirtyMovieData.map(movie => {
//     return {
//       movie_id: movie.id,
//       title: movie.title,
//       poster_path: movie.poster_path,
//       release_date: movie.release_date,
//       vote_average: movie.vote_average,
//       overview: movie.overview
//     };
//   });
//   return cleanMovieData;
// };

export const fetchMovieData = async () => {
  const rootUrl = 'https://api.themoviedb.org/3/movie/now_playing?page=1';
  try {
    const response = await fetch(
      `${rootUrl}&language=en-US&api_key=${apiKey}&language=en-US`
    );
    const nowPlayingData = await response.json();
    const cleanedMovies = cleanMovies(nowPlayingData);
    return cleanedMovies;
  } catch (error) {
    throw error;
  }
};

export const cleanMovies = movieData => {
  return movieData.results.map(movie => ({
    movie_id: movie.id,
    title: movie.title,
    poster_path: movie.poster_path,
    release_date: movie.release_date,
    vote_average: movie.vote_average,
    overview: movie.overview
  }));
};
