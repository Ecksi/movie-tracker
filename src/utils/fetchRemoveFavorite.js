export const fetchRemoveFavorite = async (user_id, movie_id) => {
  try {
    const url = `http://localhost:3000/api/users/${user_id}/favorites/${movie_id}`;
    const options = { method: 'DELETE' };
    const response = await fetch(url, options);

    if (!response.ok) {
      throw new Error(`${response.status}`);
    }
    response;
  } catch (error) {
    throw new Error(`Network request failed. (error: ${error.message})`);
  }
};
