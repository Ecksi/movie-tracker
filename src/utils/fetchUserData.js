import { doFetch } from './doFetch';

export const fetchUserData = async () =>
  await doFetch('http://localhost:3000/api/users');
