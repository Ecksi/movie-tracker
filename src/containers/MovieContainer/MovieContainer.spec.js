import React from 'react';
import { shallow, mount } from 'enzyme';
import {
  MovieContainer,
  mapStateToProps,
  mapDispatchToProps
} from './MovieContainer';
import { addMovies } from '../../actions';
import { fetchMovieData } from '../../utils/fetchMovieData';
jest.mock('../../utils/fetchMovieData');

describe('MovieContainer', () => {
  let movieContainer;
  let mockAddMovies;
  let mockProps;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {
      movies: [],
      favorites: [],
      loggedIn: false,
      userId: 1,
      addMovies: jest.fn(),
      loadFavorites: jest.fn(),
      location: {}
    };
    mockAddMovies = jest.fn();

    movieContainer = shallow(<MovieContainer {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches the snapshot', () => {
    expect(movieContainer).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls getMovieData', () => {
      const mockGetMovieData = (movieContainer.instance().getMovieData = jest.fn());
      movieContainer.instance().componentDidMount();

      expect(mockGetMovieData).toHaveBeenCalledTimes(1);
    });
  });

  describe('getMovieData', () => {
    it('calls fetchMovieData', () => {
      movieContainer.instance().getMovieData();

      expect(fetchMovieData).toHaveBeenCalledTimes(1);
    });

    it.skip('calls addMovies action', async () => {
      fetchMovieData.mockImplementation(() => 1);
      console.log(fetchMovieData);
      // console.log(movieContainer.instance().props);
      const addMoviesProp = movieContainer.instance().props.addMovies;
      console.log(addMoviesProp);
      await movieContainer.instance().getMovieData;

      expect(addMoviesProp).toHaveBeenCalledTimes(1);
    });
  });
  describe('toggleFavorite', () => {
    it('if loggedIn is true, and checkIfFavorite is true, calls removeFavorite , ', () => {
      mockProps.loggedIn = true;
      movieContainer = shallow(<MovieContainer {...mockProps} />, {
        disableLifecycleMethods: true
      });
      const checkIfFavorite = (movieContainer.instance().checkIfFavorite = jest
        .fn()
        .mockImplementation(() => true));
      const removeFavorite = (movieContainer.instance().removeFavorite = jest.fn());

      movieContainer.instance().toggleFavorite();

      expect(removeFavorite).toHaveBeenCalledTimes(1);
    });

    it('if loggedIn is true, and checkIfFavorite is false, calls addFavorite, ', () => {
      mockProps.loggedIn = true;
      movieContainer = shallow(<MovieContainer {...mockProps} />, {
        disableLifecycleMethods: true
      });
      const checkIfFavorite = (movieContainer.instance().checkIfFavorite = jest.fn());
      const addFavorite = (movieContainer.instance().addFavorite = jest.fn());
      movieContainer.instance().toggleFavorite();

      expect(checkIfFavorite).toHaveBeenCalledTimes(1);
      expect(addFavorite).toHaveBeenCalledTimes(1);
    });

    it('calls alert if user loggedIn is false', () => {
      window.alert = jest.fn();
      movieContainer.instance().toggleFavorite();

      expect(window.alert).toHaveBeenCalledTimes(1);
      expect(window.alert).toHaveBeenCalledWith(
        'You must be logged in to add favorites'
      );
    });
  });
  describe('checkIfFavorite', () => {});
  describe('addFavorite', () => {});
  describe('removeFavorite', () => {});
  describe('movieCards', () => {});
});
