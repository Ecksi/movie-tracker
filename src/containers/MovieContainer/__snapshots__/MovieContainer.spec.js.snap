// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieContainer
    addMovies={[Function]}
    favorites={Array []}
    loadFavorites={[Function]}
    location={Object {}}
    loggedIn={false}
    movies={Array []}
    userId={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "MovieContainer",
    },
    "ref": null,
    "rendered": Array [],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "MovieContainer",
      },
      "ref": null,
      "rendered": Array [],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
